Core/include/Acts/Propagator/DenseEnvironmentExtension.hpp

```diff
@@ -28,38 +30,35 @@ namespace Acts {
 /// propagation is in a TrackingVolume with attached material.
 /// @note This it templated on the floating point type because of the autodiff
 /// plugin.
-template <typename float_t>
+template <typename scalar_t>
 struct GenericDenseEnvironmentExtension {
-  /// @brief templated Vector3D replacement
-  template <typename T>
-  using Vector3D = Eigen::Matrix<T, 3, 1>;
-  /// @brief templated FreeVector replacement
-  template <typename T>
-  using FreeVector = Eigen::Matrix<T, 8, 1>;
+  using Scalar = scalar_t;
+  /// @brief Vector3D replacement for the custom scalar type
+  using ThisVector3D = Acts::ActsVector<Scalar, 3>;
 
   /// Momentum at a certain point
-  float_t currentMomentum = 0.;
+  scalar_t currentMomentum = 0.;
```

Comment:
```suggestion
  Scalar currentMomentum = 0.;
```
Nitpicking: this uses the local typedef in other parts of the codebase.